### PR TITLE
Implementation of perturb learning rate callback

### PR DESCRIPTION
**Pull request by samadejacobs**
Callback for learning rate exploration in tournament (population)-based training. Current implementation supports 
random perturbation, performed in log space.

### CODE REVIEW

File: src/callbacks/perturb_lr.cpp [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+     CEREAL_NVP(m_weights_names));
+}
+
+void perturb_lr::setup(model* m) {
+  perturb(*m);
+}
+
+void perturb_lr::on_batch_begin(model* m) {
+  const auto& c = m->get_execution_context();
+  if (m_perturb_during_training && c.get_step() > 0) {
```
Comment (timmoon10) [inline src/callbacks/perturb_lr.cpp:74]: We should only perturb once per interval:
```suggestion
  if (m_perturb_during_training
      && c.get_step() % m_batch_interval == 0
      && c.get_step() > 0) {
```

**Review by timmoon10 — CHANGES_REQUESTED** [review]
Two requests:
- Change name to "perturb_learning_rate". The meaning of LR is not obvious to me.
- Make sure we don't perturb at every step (I now see why my experiments with perturbing Adam were so volatile...).

**Comment by samadejacobs** [comment]
Two requests by @timmoon10 addressed in latest push.

**Review by timmoon10 — APPROVED** [review]
LGTM. If there aren't any new failures in Bamboo, we can merge.
